monarchmoney/monarchmoney.py

```diff
@@ -418,19 +419,175 @@ async def get_transaction_tags(self) -> Dict[str, Any]:
+              logoUrl
+              __typename
+            }
+            __typename
+          }
+          summary {
+            sumIncome
+            sumExpense
+            __typename
+          }
+          __typename
+        }
+        summary: aggregates(filters: $filters, fillEmptyValues: true) {
+          summary {
+            sumIncome
+            sumExpense
+            savings
+            savingsRate
+            __typename
+          }
+          __typename
+        }
+      }
+    """)
+
+    variables = {
+      "limit": 25,
+      "orderBy": "date",
+      "filters": {
+        "search": "",
+        "categories": [],
+        "accounts": [],
+        "tags": [],
+      }
+    }
+
+    if start_date and end_date:
+      variables["filters"]["startDate"] = start_date
+      variables["filters"]["endDate"] = end_date
+    elif bool(start_date) != bool(end_date):
+      raise Exception("You must specify both a startDate and endDate, not just one of them.")
+    else:
+      current_year = datetime.now().year
+      current_month = datetime.now().month
+      last_date = calendar.monthrange(current_year, current_month)[1]
+      variables["filters"]["startDate"] = datetime(current_year, current_month, 1).strftime("%Y-%m-%d")
+      variables["filters"]["endDate"] = datetime(datetime.now().year, datetime.now().month, last_date).strftime("%Y-%m-%d")
+
+    return await self.gql_call(
+        operation="Web_GetCashFlowPage",
+        variables=variables,
+        graphql_query=query
+    )
+
+  async def get_cashflow_summary(self,
+      start_date: Optional[str]=None,
+      end_date: Optional[str]=None
+    ) -> Dict[str, Any]:
+    """
+    Gets all the categories configured in the account.
+    """
+    query = gql("""
+      query Web_GetCashFlowPage($filters: TransactionFilterInput) {
+        summary: aggregates(filters: $filters, fillEmptyValues: true) {
+          summary {
+            sumIncome
+            sumExpense
+            savings
+            savingsRate
+            __typename
+          }
+          __typename
+        }
+      }
+    """)
+
+    variables = {
+      "limit": 25,
+      "orderBy": "date",
+      "filters": {
+        "search": "",
+        "categories": [],
+        "accounts": [],
+        "tags": [],
+      }
+    }
+
+    if start_date and end_date:
+      variables["filters"]["startDate"] = start_date
+      variables["filters"]["endDate"] = end_date
+    elif bool(start_date) != bool(end_date):
+      raise Exception("You must specify both a startDate and endDate, not just one of them.")
+    else:
+      current_year = datetime.now().year
+      current_month = datetime.now().month
+      last_date = calendar.monthrange(current_year, current_month)[1]
+      variables["filters"]["startDate"] = datetime(current_year, current_month, 1).strftime("%Y-%m-%d")
+      variables["filters"]["endDate"] = datetime(datetime.now().year, datetime.now().month, last_date).strftime("%Y-%m-%d")
```

Comment:
same question about using a future date

```suggestion
      variables["filters"]["endDate"] = datetime(current_year,  current_month, last_date).strftime("%Y-%m-%d")
```

Reply:
Yes, this is how it's queried on the website for current month